### ISSUE TITLE
AWS::Organizations::OrganizationalUnit support resource import

### ISSUE THREAD

**Issue by joneskoo** [aws-cloudformation/cloudformation-coverage-roadmap#1411]
### Name of the resource

Other

### Resource name

AWS::Organizations::OrganizationalUnit

### Description

#33 added support for Organizations resources. To start using this with exisiting Organization and manage OUs as code, I would need resource import to support OrganizationalUnit and ideally also Policy and Account resources. However for OU there is no workaround as moving accounts between OUs would cause possibly risky/disruptive triggering of CloudFormation stackset operations. That’s why it’s important to have native resource import support.

### Other Details

#33

**Comment by sjegannath** [comment]
Is this related to https://www.chrisfarris.com/post/organization-cloudformation/ ?

**Comment by joneskoo** [comment]
Not related in the sense that I'd be using org-formation or had read that.

Interestingly that article says they were able to import AWS::Organizations::Account despite documentation implying you can't; these resources are not listed as being importable.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import.html

So is this only a documentation issue or is importing these resources unsupported?

**Comment by tennex-adam** [comment]
> Interestingly that article says they were able to import AWS::Organizations::Account despite documentation implying you can't; these resources are not listed as being importable.

I've imported several accounts in `us-east-1` without issue.  It's worth mentioning that I moved the accounts from their respective OUs to the Org root, imported the accounts, then re-assigned them to their respective OUs.  An import with an existing OU assignment may work, but I didn't test it.

> So is this only a documentation issue or is importing these resources unsupported?

I'm curious as well.  Since CloudFormation has released the ability to import the [documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import-supported-resources.html) hasn't always lined up with my success rate.  I'll generally try an import if the value of having the resource folded into the codebase (CDK) can deprecate a custom resource.

**Comment by reidca** [comment]
This would be a very useful feature to establish IaC control over existing AWS Organizations

**Comment by joneskoo** [comment]
I have successfully imported OUs, SCPs and Tag Policies to CloudFormation. I consider this one resolved.

There was two minor issues which I reported via support, those might or might not be fixed already but import did work. Take care to specify the correct parent.

**Comment by reidca** [comment]
Sadly this does not work when importing more than one or two OUS. Having followed up with AWS support and ultimately the CloudFormation and AWS Organizations teams, it appears that cfn calls the aws organizations apis too many times during the import process and is rate limited and then fails. I am told a fix is on the way but that was months ago.

**Comment by joneskoo** [comment]
I hit the same rate limiting but was ultimately successful. I had to scratch my head many times because import didn't validate the parent id was correct on the imported resource and that caused it to try to update it, hitting rate limit but not actually changing anything.

In any case, I was able to import, at least incrementally adding more resources to the imported stack. I don't think I saw it being rate limited when the parent was correctly set to match the physical resources in my case, but I believe you if it still happens. Good luck.

In any case the feature is there and the discussion belongs to support or new issue.
